src/fmpz_mat/pow.c

```diff
@@ -46,7 +46,7 @@ fmpz_mat_pow(fmpz_mat_t B, const fmpz_mat_t A, ulong exp)
         fmpz_mat_init_set(T, A);
         fmpz_mat_init(U, d, d);
 
-        for (i = ((slong) FLINT_BIT_COUNT(exp)) - 2; i >= 0; i--)
+        for (i = FLINT_BIT_COUNT(exp) - 2; i >= 0; i--)
```

Comment:
The cast was removed on purpose?

Reply:
by request of @albinahlback in https://github.com/flintlib/flint/pull/2189#discussion_r1930273351